src/R/Support/Impl/Help/IntelliSenseRHost.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Languages.Editor.Shell;
+using Microsoft.R.Host.Client;
+using Microsoft.R.Support.Settings;
+
+namespace Microsoft.R.Support.Help {
+    [Export(typeof(IIntellisenseRHost))]
+    public sealed class IntelliSenseRHost : IIntellisenseRHost {
+        private static readonly Guid SessionId = new Guid("8BEF9C06-39DC-4A64-B7F3-0C68353362C9");
+        private readonly ICoreShell _coreShell;
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly SemaphoreSlim _sessionSemaphore = new SemaphoreSlim(1, 1);
+
+        [ImportingConstructor]
+        public IntelliSenseRHost(ICoreShell coreShell, IRSessionProvider sessionProvider) {
```

Comment:
My question was more about state of the session. `IRSessionProvider.BeginEvaluationAsync` stops the host after evaluation is released, so if there should be some preloaded data or preserved state from previous calls, that's not an option.

---
In reply to: [71205046](https://github.com/Microsoft/RTVS/pull/2084#discussion_r71205046) [](ancestors = 71205046)